core-feature-pack/common/src/main/resources/modules/system/layers/base/org/bouncycastle/bcutil/main/module.xml

```diff
@@ -0,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~
+  ~ Copyright 2020 Red Hat, Inc., and individual contributors
+  ~ as indicated by the @author tags.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~     http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<module xmlns="urn:jboss:module:1.9" name="org.bouncycastle.bcutil">
```

Comment:
Thanks for your feedback. I'll fix them all.

> I've added some details that need to be addressed for merging this upgrade. In addition, the Full integration - Linux is also failing, I haven't checked the root reason yet but we need to understand why it is failing as well.

Full integration fails due to bcmail:
[https://mvnrepository.com/artifact/org.bouncycastle/bcmail-jdk15on/1.69](https://mvnrepository.com/artifact/org.bouncycastle/bcmail-jdk15on/1.69) because it needs bcutil.

Reply:
You will need to exclude `org.bouncycastle:bcutil-jdk15on` from the `bcmail-jdk15on` and `bcpkix-jdk15on` maven dependencies declared in the wildfly-core/pom.xml. The  Full integration - Linux Job is still failing because those dependencies are being pulled in from wildfly-core.

Reply:
@yersan Maybe I don't understand the dependency but [WFLY-14880](https://issues.redhat.com/browse/WFLY-14880) PR [https://github.com/wildfly/wildfly/pull/14357](https://github.com/wildfly/wildfly/pull/14357) addresses [bcmail-jdk15on](https://mvnrepository.com/artifact/org.bouncycastle/bcmail-jdk15on/1.69) with the new bcutil-jdk15on.

Or do you request a different change approach: 1) Exclude bcutil-jdk15on from the bcmail-jdk15on in WFLY 2) Merge this PR 3) After WFCORE merge in WFLY undo step 1 and include bcutil-jdk15on ?

Reply:
The issue here is the failing test for this PR 'Full integration - Linux'. This Job tests the wildfly-core with your changes with wildlfy full (master branch):
```
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.BanTransitiveDependencies failed with message:
[08:43:25]	org.wildfly:wildfly-ee-feature-pack-common:pom:25.0.0.Beta1-SNAPSHOT
[08:43:25]	   org.bouncycastle:bcmail-jdk15on:jar:1.69:compile has transitive dependencies:
[08:43:25]	      org.bouncycastle:bcutil-jdk15on:jar:1.69:compile
[08:43:25]	
[08:43:25]	Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M2:enforce (ban-transitive-deps) on project wildfly-ee-feature-pack-common: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed.
[08:43:50]	
```

Notice that WFLY-14880 declare the maven dependencies without specifying the version, the  bcmail-jdk15on dependency is pulled in by the wildfly-core maven dependency, https://github.com/wildfly/wildfly/blob/master/pom.xml#L491-L497

However, there are rules to control how transitive dependencies are pulled in, hence the Job is complaining about it because bcmail-jdk15on has a transitive dependency on bcutil-jdk15on.

So my understanding is we need to exclude bcutil-jdk15on transitive dependency on the bcmail-jdk15on dependency declared in wildfly-core/pom.xml.

An example of exclusions is here: https://github.com/wildfly/wildfly-core/blob/main/pom.xml#L1277-L1282

Reply:
Thank you for the explanation. I have added the exclusion.

Reply:
It seems I have to provide the bcutil separate at other points now:
```
[02:17:42] [Step 2/3] [ERROR] org.wildfly.extension.elytron.TlsTestCase  Time elapsed: 0.002 s  <<< ERROR!
[02:17:42] [Step 2/3] java.lang.NullPointerException
[02:17:42] [Step 2/3] 	at org.wildfly.extension.elytron.TlsTestCase.cleanUpTests(TlsTestCase.java:340)
[02:17:42] [Step 2/3] 
[02:17:43] [Step 2/3] org.wildfly.extension.elytron.TlsTestCase
[02:17:43] [org.wildfly.extension.elytron.TlsTestCase] TlsTestCase
[02:17:43] [TlsTestCase] java.lang.NoClassDefFoundError: org/bouncycastle/asn1/cms/CMSObjectIdentifiers
[02:17:43] [TlsTestCase] java.lang.NoClassDefFoundError: org/bouncycastle/asn1/cms/CMSObjectIdentifiers
	at org.wildfly.extension.elytron.TlsTestCase.createCRL(TlsTestCase.java:231)
	at org.wildfly.extension.elytron.TlsTestCase.setUpKeyStores(TlsTestCase.java:284)
	at org.wildfly.extension.elytron.TlsTestCase.initTests(TlsTestCase.java:326)
Caused by: java.lang.ClassNotFoundException: org.bouncycastle.asn1.cms.CMSObjectIdentifiers
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 3 more
[02:17:43] [org.wildfly.extension.elytron.TlsTestCase] TlsTestCase
[02:17:43] [TlsTestCase] java.lang.NullPointerException
[02:17:43] [TlsTestCase] java.lang.NullPointerException
	at org.wildfly.extension.elytron.TlsTestCase.cleanUpTests(TlsTestCase.java:340)
```

```
grep --include=\pom.xml -rnw '/wildfly-core/' -e "bcpkix-jdk15on"
/wildfly-core/testsuite/standalone/pom.xml:100:            <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/core-feature-pack/common/pom.xml:237:            <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/pom.xml:1219:                        <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/pom.xml:1307:                <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/testbom/pom.xml:177:                        <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/server/pom.xml:223:            <artifactId>bcpkix-jdk15on</artifactId>
/wildfly-core/elytron/pom.xml:271:            <artifactId>bcpkix-jdk15on</artifactId>
```